File: Data Structures & Algorithms/search-insert-position/submission-2.java
class Solution {
    public int searchInsert(int[] nums, int target) {
     int left = 0; 
     int right = nums.length - 1;
     int mid = (right + left) / 2;

     while (left <= right){
        if (nums[mid] == target){
            return mid;
        }

        if(nums[mid] > target){
            right = mid - 1;
        }

        if (nums[mid] < target){
            left = mid + 1;
        }

        mid = (right + left) / 2;
     }

        return left;
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        # expand window until we find subarray sum == target
        # shrink window sum -= nums[left] 
        # expand again to right
        # res = min(minimalSoFar, currentWindowSize)

        res = float('inf')
        left = 0
        curSum = 0

        for right, num in enumerate(nums):
            curSum += num

            while(curSum >= target):
                res = min(right-left + 1, res)
                curSum -= nums[left]
                left += 1

        if res == float('inf'):
            return 0

        
        return res


